### ISSUE TITLE
`no-unnecessary-string-trim` incorrectly flags Zod `.trim()` method as unnecessary before `.startsWith()`

### ISSUE THREAD

**Issue by timdavish** [sindresorhus/eslint-plugin-unicorn#3576]
The `no-unnecessary-string-trim` rule incorrectly flags `.trim()` calls on Zod schemas (`ZodString`) as unnecessary when chained before `.startsWith()`. Zod's `.trim()` is not `String.prototype.trim()` being called on an already-known string value. The rule appears to match purely on method name without verifying that the target is actually a string.

Rule: `no-unnecessary-string-trim`

This incorrectly triggers the rule:

```ts
import {z} from "zod"

const schema = z.string().trim().startsWith("asdf")
```
